### ISSUE TITLE
[DL3025] False positive with multiline instruction

### ISSUE THREAD

**Issue by CodingKoopa** [hadolint/hadolint#783]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
hadolint permits the user to have a multi-line ENTRYPOINT or CMD, using *exec* form.

### Actual behavior
hadolint raises DL3025 when the instruction is made to span multiple lines by escaping the newline.

### Steps to reproduce the behavior
These do not raise an error, and are well-formed (that is, no issue here):
```Dockerfile
CMD [ "/bin/sh", "-c", "true;"]
```
```Dockerfile
CMD ["/bin/sh", "-c", \
  "true;"]
```
These do raise `DL3025`, but I think they are well-formed:
```Dockerfile
CMD ["/bin/sh", "-c", \
  "true; \
  true;"]
```
```Dockerfile
CMD ["/bin/sh", "-c", \
  "true && \
  true"]
```

These statements raise `DL3025`, but also are ill-formed for unrelated reasons:
```Dockerfile
CMD ["/bin/sh", "-c", \
  ]
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter UNKNOWN
```
(what I have installed is [hadolint-bin](https://aur.archlinux.org/packages/hadolint-bin) 2.8.0-1 from the AUR.)

Thanks!

**Comment by m-ildefons** [comment]
Hi @CodingKoopa ,

the _exec_ form in `CMD`, `RUN` and `ENTRYPOINT` statements are parsed as JSON arrays. As far as I can see the examples given that raise the error are not well formed JSON:
```shell
│ ~/tmp │> cat -n longline.json 
     1  [
     2    "/bin/bash",
     3    "-c",
     4    "true; \
     5    true"
     6  ]
│ ~/tmp │> cat longline.json | jq
parse error: Invalid escape at line 5, column 7
```
while not breaking the string at line 4/5 is valid JSON:
```shell
│ ~/tmp │> cat -n longline.json 
     1  [
     2    "/bin/bash",
     3    "-c",
     4    "true; true"
     5  ]
│ ~/tmp │> cat longline.json | jq
[
  "/bin/bash",
  "-c",
  "true; true"
]

```
The same applies to the second example. The third one is invalid because there is a superfluous `,` after the `"-c"` and works just fine without it for me:
```shell
│ ~/tmp │> cat -n Dockerfile
     1  FROM alpine:3.14
     2  CMD [ "/bin/bash", "-c" \
     3      ]
│ ~/tmp │> hadolint Dockerfile
│ ~/tmp │> 
```

Weirdly the ones that are not well formed JSON on their own build just fine when used in a Dockerfile. What's even more weird is that escaping the escape `\` works fine in Hadolint then, but breaks in Docker:
```Dockerfile
CMD [ "/bin/sh", "-c", \
      "echo foo && \\
      echo bar" ]
```
So yeah, there is definitively something funky going on in the parser. Thanks for reporting.

**Comment by CodingKoopa** [comment]
Thanks for the quick turnaround!
